client/src/nl/sensorlab/videowall/animation/baseanimations/flocking/Boid.java

```diff
@@ -0,0 +1,190 @@
+package nl.sensorlab.videowall.animation.baseanimations.flocking;
+
+import java.util.ArrayList;
+
+import nl.sensorlab.videowall.animation.BaseAnimation;
+import processing.core.PApplet;
+import processing.core.PGraphics;
+import processing.core.PVector;
+
+public class Boid {
+
+	private PVector position;
+	private PVector velocity;
+	private PVector acceleration;
+
+
+	public float radius = 0.5f;
+	public float maxSpeed = 0.35f;
+	public float maxForce = 0.355f;
+	public float neighbourDistance = 20;
+
+	int color;
+
+	public Boid(float x ,float y, int color) {
+		this.position = new PVector(x, y);
+		this.acceleration = new PVector(0,0);
+		this.color = color;
+
+		float angle = (float)(Math.random() * (Math.PI * 2));
+		this.velocity = new PVector((float)Math.cos(angle), (float)Math.sin(angle)); 
+	}
+
+	public void update(ArrayList<Boid> boids) {
+		flock(boids);
+
+		// Update velocity
+		velocity.add(acceleration);
+
+		// Limit speed
+		velocity.limit(maxSpeed);
+		position.add(velocity);
+
+		// Reset accelertion to 0 each cycle
+		acceleration.mult(0);
+
+		// Keep the boid within the window
+		keepInBounds();
+	}
+
+	private void flock(ArrayList<Boid> boids) {	
+		PVector sep = separate(boids);   // Separation
+		PVector ali = align(boids);      // Alignment
+		PVector coh = cohesion(boids);   // Cohesion
+
+		// Arbitrarily weight these forces
+		sep.mult(1.5f);
+		ali.mult(1.0f);
+		coh.mult(1.0f);
+
+		// Add the force vectors to acceleration
+		acceleration.add(sep);
+		acceleration.add(ali);
+		acceleration.add(coh);
+	}
+
+	public void keepInBounds() {
+		if (position.x < - radius) {
```

Comment:
Is it possible to have both 'x' conditions true? Otherwise you can save a little bit of performance by using an else if statement for the other 'x' condition. Same goes for y